### ISSUE TITLE
[REVIEW]: tbeptools: An R package for synthesizing estuarine data for environmental research

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3485]
**Submitting author:** @fawda123 (<a href="http://orcid.org/0000-0002-4996-0059">Marcus Beck</a>)
**Repository:** <a href="https://github.com/tbep-tech/tbeptools" target ="_blank">https://github.com/tbep-tech/tbeptools</a>
**Version:** v1.1.0
**Editor:** @kthyng
**Reviewer:** @paleolimbot, @richardsc
**Archive:** <a href="https://doi.org/10.5281/zenodo.5514034" target="_blank">10.5281/zenodo.5514034</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/2fc39b299e66e93f1dbb05a9d57117ff/status.svg)](https://joss.theoj.org/papers/2fc39b299e66e93f1dbb05a9d57117ff)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/2fc39b299e66e93f1dbb05a9d57117ff"><img src="https://joss.theoj.org/papers/2fc39b299e66e93f1dbb05a9d57117ff/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/2fc39b299e66e93f1dbb05a9d57117ff/status.svg)](https://joss.theoj.org/papers/2fc39b299e66e93f1dbb05a9d57117ff)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/tbep-tech/tbeptools" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@paleolimbot & @richardsc, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kthyng know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/tbep-tech/tbeptools">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@fawda123) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/tbep-tech/tbeptools">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@fawda123) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @paleolimbot, @richardsc it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.18 s (861.7 files/s, 53824.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
R                              134            931           2197           4131
XML                              2              0              3            454
HTML                             1             90              5            353
TeX                              6              6              0            250
YAML                             5             37              6            214
Rmd                              5            290            486            206
Markdown                         5             54              0            156
-------------------------------------------------------------------------------
SUM:                           158           1408           2697           5764
-------------------------------------------------------------------------------


Statistical information for the repository '47767fb76ea97795cd0cc7b7' was
gathered on 2021/07/12.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Ben Best                         3          4956              0            4.82
fawda123                        25         46437          51393           95.18

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.ecolind.2020.107310 is OK
- 10.1353/sgo.2017.0026 is OK

MISSING DOIs

- 10.1007/s12237-021-00974-7 may be a valid DOI for title: Developing a water quality assessment framework for southwest Florida tidal creeks

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03485/joss.03485/10.21105.joss.03485.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03485/joss.03485/10.21105.joss.03485.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @richardsc, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @paleolimbot, please update us on how your review is going (this is an automated reminder).

**Comment by paleolimbot** [comment]
My first round of comments are in as issues in the target repo! It's an excellent example of well-tested code wrapped up as a package to support automated reporting of environmental data and will be an excellent contribution to JOSS!

The three things I'd like to see are (1) a tiny bit of example usage in the paper, (2) updating the Statement of Need section to mention the Shiny app and the role that this package fills in the reporting component, and (3) a slightly broader scope when reviewing previous work. Ultimately I'd like the paper to speak to the bigger problems that other research software writers need to solve outside the scope of the Tampa Bay Estuary...those three bullet points are my two cents on what that is although I'm happy to entertain other ideas!

**Comment by fawda123** [comment]
Awesome, thanks @paleolimbot!  I really appreciate the comments.

**Comment by richardsc** [comment]
Sorry, I'm running a bit behind. Plan to get to it this weekend!

**Comment by kthyng** [comment]
@fawda123 Have you been able to address any of the input from @paleolimbot?

@richardsc Will you be able to start your review soon?

**Comment by fawda123** [comment]
@kthyng thanks for checking in, no I have not yet addressed any of the comments from @paleolimbot.  I was waiting for review from @richardsc, but can proceed with revision this wek if preferred.

**Comment by fawda123** [comment]
@kthyng, reviews from @paleolimbot have been addressed in the revision.  Waiting on reviews from @richardsc.

**Comment by richardsc** [comment]
Sorry for the delay in completing this. Summer, pandemic, etc etc ... 

I had the benefit of reviewing after @paleolimbot , and my early review comments mirrored many of his. The paper is much improved by the changes since the first draft, and I think is acceptable for publication. I have two points for consideration, but these are not blocking in any way.

1. I made an issue on the repo (tbep-tech/tbeptools#47) to consider using more consistent and verbose function names. This is partly personal style preference, but there is so much functionalilty in the package (which is great!) that it is a bit hard to intuit what all the abbreviations mean.

2. Similar to @paleolimbot, I think a discussion of how this package could be used and/or generalized to be more broadly useful to the NEP community (or really *any* estuarine scientist) would improve the scope of the paper. I don't think that generalization has to be done here (it is a big undertaking), but would strengthen the potential utility of the package beyond users specifically interested in *only* the Tampa Bay Estuary. This point is touched on in the current version (including pointing out the `peptools` package), but I think it could be expounded upon a little further. Does every estuary need its own package to handle data import and analysis/reporting? The answer could be "yes", which is ok, but some thought about a broader approach would be nice.

All said, this is a nice paper and a great package. Well done @fawda123!

**Comment by fawda123** [comment]
Thanks @richardsc, I really appreciate the comments.  I agree some of the function names could be more intuitive and I will work on some text revisions to discuss generalization/needs of other estuarine scientists.

**Comment by paleolimbot** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03485/joss.03485/10.21105.joss.03485.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03485/joss.03485/10.21105.joss.03485.pdf) :page_facing_up: :point_left:

**Comment by paleolimbot** [comment]
I just did a read-through on the paper and revised README and like the changes. No further acceptance blockers from me, although I would put an `# Example` header before your example since it's not really in the `# Statement of need` section.

**Comment by fawda123** [comment]
@paleolimbot I just added an `# Example usage` section https://github.com/tbep-tech/tbeptools/commit/ec595837d30d278c2fd1cb2a338eb76f34594f94

**Comment by fawda123** [comment]
@kthyng @richardsc @paleolimbot I'm done addressing the comments from all.  I really appreciate the thoughtful comments on this package and paper.  I think we've got a much better product as a result. 

For the first comment from @richardsc, I've added some descriptive text describing the suffixes (see my response here https://github.com/tbep-tech/tbeptools/issues/47).  For the second comment (i.e., "A discussion how the package can be used beyond the NEPs..."), I feel that the revisions in response @paleolimbot address that concern (see the commit here https://github.com/tbep-tech/tbeptools/commit/227915bc7024de827f3f2f83bf482db7a82755d1).  I'm happy to revise further, if needed, but I am done for now.  All, please let me know if these changes are sufficient.

**Comment by paleolimbot** [comment]
I'm happy with this! Cheers!

**Comment by richardsc** [comment]
I am also happy with the recent changes.

**Comment by kthyng** [comment]
Excellent! Thanks so much for everyone's work on this and good job @fawda123! I'll work on the final steps of this process.

**Comment by kthyng** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5281/zenodo.3648664 is OK
- 10.5281/zenodo.4319936 is OK
- 10.5281/zenodo.3939236 is OK
- 10.5281/zenodo.3701560 is OK
- 10.5066/P9X4L3GE is OK
- 10.1016/j.ecolind.2020.107310 is OK
- 10.1353/sgo.2017.0026 is OK
- 10.1007/s12237-021-00974-7 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03485/joss.03485/10.21105.joss.03485.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03485/joss.03485/10.21105.joss.03485.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
Paper looks good!

**Comment by kthyng** [comment]
@fawda123 Can you verify what version of the software should be associated with your JOSS publication? Also, can you make an archive of the software at a place like Zenodo and then report the doi here to associate with your paper? Please modify the metadata of the archive as needed so that the title and author list exactly match your JOSS paper.

**Comment by fawda123** [comment]
@kthyng v1.1.0 should be used.  DOI link on Zenodo is [https://doi.org/10.5281/zenodo.5514034](https://zenodo.org/badge/latestdoi/184627857), w/ matching title and author list for the paper.

**Comment by kthyng** [comment]
@whedon set v1.1.0 as version

**Comment by whedon** [comment]
OK. v1.1.0 is the version.

**Comment by kthyng** [comment]
@whedon set 10.5281/zenodo.5514034 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5514034" target="_blank">10.5281/zenodo.5514034</a> is the archive.

**Comment by kthyng** [comment]
Ok all set!

**Comment by kthyng** [comment]
@whedon accept deposit=True

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1438948107660800008) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2589
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03485](https://doi.org/10.21105/joss.03485)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congratulations on your new publication @fawda123! Many thanks to reviewers @paleolimbot and @richardsc for your time, hard work, and expertise!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03485/status.svg)](https://doi.org/10.21105/joss.03485)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03485">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03485/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03485/status.svg
   :target: https://doi.org/10.21105/joss.03485
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03485/status.svg)](https://doi.org/10.21105/joss.03485)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by fawda123** [comment]
Thanks again @kthyng for shepherding this through and thanks of course to @richardsc and @paleolimbot for the help.  Will def sign up to review in the future.

**Comment by kthyng** [comment]
@fawda123 https://joss.theoj.org/reviewer-signup.html
